Userland/Utilities/hostname.cpp

```diff
@@ -5,27 +5,23 @@
  */
 
 #include <LibCore/ArgsParser.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
 #include <limits.h>
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
 
-int main(int argc, char** argv)
+ErrorOr<int> serenity_main(Main::Arguments args)
 {
     const char* hostname = nullptr;
 
     Core::ArgsParser args_parser;
     args_parser.add_positional_argument(hostname, "Hostname to set", "hostname", Core::ArgsParser::Required::No);
-    args_parser.parse(argc, argv);
+    args_parser.parse(args);
 
     if (!hostname) {
-        char buffer[HOST_NAME_MAX];
-        int rc = gethostname(buffer, sizeof(buffer));
-        if (rc < 0) {
-            perror("gethostname");
-            return 1;
-        }
-        outln("{}", buffer);
+        outln("{}", TRY(Core::System::gethostname()));
     } else {
         if (strlen(hostname) >= HOST_NAME_MAX) {
             warnln("Hostname must be less than {} characters", HOST_NAME_MAX);
```

Comment:
if Core::System::sethostname does not exist, please add it... usually i find all the places that use perror, and try to remove all of them, even if i need to add support to LibCore/System.{h,cpp}